mono-traversable/src/Data/MonoTraversable.hs

```diff
@@ -1382,3 +1384,27 @@ ointercalate x = mconcat . List.intersperse x . otoList
         ointercalate x = T.intercalate x . otoList #-}
 {-# RULES "intercalate LText" forall x.
         ointercalate x = TL.intercalate x . otoList #-}
+
+-- | Provides a `MonoFoldable`, `MonoFunctor` or `MonoPointed` for an arbitrary
+-- `F.Foldable`, `Functor` or `Applicative`.
+--
+-- Useful for, e.g., passing a `F.Foldable` type you don't own into a
+-- function that expects a `MonoFoldable`.
+--
+-- > // package A
+-- > data MyList a = MyList [a] deriving Foldable
+-- >
+-- > // package B
+-- > process :: MonoFoldable mono => mono -> IO ()
+-- >
+-- > // package C
+-- > process (WrappedMono (MyList []))
+--
+-- @since 1.0.13.0
+newtype WrappedMono f a = WrappedMono { unwrapMono :: f a }
+    deriving newtype (F.Foldable, Functor, Applicative, Monad)
+
+type instance Element (WrappedMono f a) = a
+instance F.Foldable f  => MonoFoldable (WrappedMono f a)
```

Comment:
Sorry, I think I reversed things in my brain previously. `WrappedMonad` takes a type with an existing `Monad` instance and creates an `Applicative` instance. This type is doing the opposite: taking something with a polymorphic instance (e.g. `Foldable`) and creating a monomorphic instance (e.g. `MonoFoldable`). I'm now second-guessing my naming choice, and thinking it should be `WrappedPoly`. Though perhaps this type is useful to have as well.

Reply:
Makes sense, I've renamed `WrappedMono` to `WrappedPoly`!

Regarding a possible `WrappedMono`, I was playing around with that idea and this is what I have so far:

```haskell
{-# LANGUAGE GADTs                   #-}
{-# LANGUAGE InstanceSigs            #-}

data WrappedMono mono a where
    WrappedMono :: Element mono ~ a => mono -> WrappedMono mono a


unwrapMono :: WrappedMono mono a -> mono
unwrapMono (WrappedMono mono) = mono

instance MonoFoldable mono => F.Foldable (WrappedMono mono) where
    foldr :: (a -> b -> b) -> b -> WrappedMono mono a -> b
    foldr f zero (WrappedMono xs) = ofoldr f zero xs
```

First thing I did was tag `WrappedMono mono` with its element type `a`, because foldables have to be of kind `* -> *`.

Second, as far as I can tell, in order for that last line to work, I need to prove that `Element mono ~ a`, so I decided to package up such proof into the `WrappedMono` constructor.
This is not ideal because it means we can't use newtype deriving anymore, but I don't see a way around having to do this.

What do you think?

If the idea seems worth pursuing, I can explore it and add it to this PR or create a separate PR, whichever you prefer.

Reply:
Since this one isn't driven by an actual user need, I'm not too concerned about it. If you think it would be useful, we can follow up in a separate PR. I don't want to block your work here on something speculative I just brought up :)